pkg/network/ebpf/c/runtime/tracer.c

```diff
@@ -145,18 +145,23 @@ static __always_inline int read_conn_tuple(conn_tuple_t* t, struct sock* skp, u6
     return read_conn_tuple_partial(t, skp, pid_tgid, type);
 }
 
-static __always_inline void handle_tcp_stats(conn_tuple_t* t, struct sock* skp) {
+static __always_inline void handle_tcp_stats(conn_tuple_t* t, struct sock* skp, __u32* segs_in, __u32* segs_out) {
     __u32 rtt = 0;
     __u32 rtt_var = 0;
     bpf_probe_read(&rtt, sizeof(rtt), &tcp_sk(skp)->srtt_us);
     bpf_probe_read(&rtt_var, sizeof(rtt_var), &tcp_sk(skp)->mdev_us);
 
+    bpf_probe_read(segs_out, sizeof(*segs_out), &tcp_sk(skp)->segs_out);
```

Comment:
Since the segment count is not a TCP-only stat, and since we are not storing it in `tcp_stats_t`, I don't think we should modify `handle_tcp_stats`. I'd put these reads in their respective probes (`tcp_sendmsg` and `tcp_cleanup_rbuf` I believe).

Reply:
I agree.  My "logic" for putting it here is that the data we're after is in the `tcp_sk`; it's a tcp-specific stat (even though we collect it/store it generally at the connection level; UDP we gather it differently).  So my thought was to only ever access the `tcp_sk` in the tcp specific code.

I'd be happy to move it around, but that was my thinking.

Reply:
Yeah, I think it is more confusing to have it here since it isn't stored/used here. The kprobes are already TCP-specific, so I think it is OK to have this code in them.